private/bufpkg/bufmodule/errors.go

```diff
@@ -209,3 +212,47 @@ func (n *NoProtoFilesError) Error() string {
 	_, _ = builder.WriteString(` had no .proto files`)
 	return builder.String()
 }
+
+// DigestMismatchError is the error returned if the Digest of a downloaded Module or Commit
+// does not match the expected digest in a buf.lock file.
+type DigestMismatchError struct {
+	ModuleFullName ModuleFullName
+	CommitID       uuid.UUID
+	ExpectedDigest Digest
+	ActualDigest   Digest
+}
+
+// Error implements the error interface.
+func (m *DigestMismatchError) Error() string {
+	if m == nil {
+		return ""
+	}
+	var builder strings.Builder
+	_, _ = builder.WriteString(`*** Digest verification failed`)
+	if m.ModuleFullName != nil {
+		_, _ = builder.WriteString(` for "`)
+		_, _ = builder.WriteString(m.ModuleFullName.String())
+		if !m.CommitID.IsNil() {
+			_, _ = builder.WriteString(`:`)
+			_, _ = builder.WriteString(uuidutil.ToDashless(m.CommitID))
+		}
+		_, _ = builder.WriteString(`"`)
+	}
+	_, _ = builder.WriteString(` ***`)
+	_, _ = builder.WriteString("\n")
+	if m.ExpectedDigest != nil && m.ActualDigest != nil {
+		_, _ = builder.WriteString("\t")
+		_, _ = builder.WriteString(`Expected digest: "`)
+		_, _ = builder.WriteString(m.ExpectedDigest.String())
+		_, _ = builder.WriteString(`"`)
+		_, _ = builder.WriteString("\n")
+		_, _ = builder.WriteString("\t")
+		_, _ = builder.WriteString(`Actual digest: "`)
+		_, _ = builder.WriteString(m.ActualDigest.String())
+		_, _ = builder.WriteString(`"`)
+		_, _ = builder.WriteString("\n")
+	}
+	_, _ = builder.WriteString("\t")
+	_, _ = builder.WriteString(`This may either be a result of an attack, or a result of your local cache being corrupted. To clear your local cache, run "buf registry cc".`)
```

Comment:
Of the two it is more likely to be the second also this could be the result of a hand edited buf.lock file.

We may also want to consider at some point allowing to clear the cache for a single module or commit instead of wiping everything.